logic/conversion/playerToLeaderboardEntry.go

```diff
@@ -8,29 +8,34 @@ import (
 	"github.com/fluofoxxo/outrun/obj"
 )
 
-func PlayerToLeaderboardEntry(player netobj.Player) obj.LeaderboardEntry {
-	friendID := player.ID // TODO: is this right?
+func PlayerToLeaderboardEntry(player netobj.Player, mode int64) obj.LeaderboardEntry {
+	friendID := player.ID
 	name := player.Username
 	url := player.Username + "_findme" // TODO: only used for testing right now
-	grade := player.PlayerState.Rank   // TODO: _probably_ what this is, but unsure
+	grade := int64(1) // TODO: make this specifiable in the future (this is the high score ranking it seems)
 	exposeOnline := int64(0)
-	rankingScore := player.PlayerState.HighScore // TODO: this probably differs based on mode...
-	rankChanged := int64(2)
+	rankingScore := player.PlayerState.HighScore
+	rankChanged := int64(0)
 	isSentEnergy := int64(0)
-	expireTime := time.Now().UTC().Unix() + 12345
-	numRank := player.PlayerState.Rank + 1
+	expireTime := now.EndOfWeek().UTC().Unix()
+	numRank := player.PlayerState.Rank
 	loginTime := player.LastLogin
 	mainCharaID := player.PlayerState.MainCharaID
-	mainCharaLevel := int64(12) // TODO: remove testing
+	mainCharaLevel := player.CharacterState[0].Level // TODO: is this right?
```

Comment:
This is incorrect. The main character is fetched by getting the index of the main character by `player.PlayerState.MainCharaID`. The player struct has a helper function for this: `player.IndexOfChara(player.PlayerState.MainCharaID)`.
Likewise, this applies to line 26, and lines 28 and 30 with `player.IndexOfChao(player.PlayerState.MainChaoID)`.

Reply:
I'll keep that in mind.